src/components/button/button.tsx

```diff
@@ -141,7 +140,13 @@ export const Button: FC<ButtonProps> = ({
       <ElButton
         data-variant={variant}
         className={combinedClassName}
-        onClick={handleButtonClick(onClick)}
+        onClick={(e) => {
+          if (!disabled) {
+            handleButtonClick(onClick)(e)
```

Comment:
**quibble:** Can we just do `onClick?.(e)` here instead of the indirection involved with `handleButtonClick`?

Reply:
@kurtdoherty I have centralized the `onClick` logic into a separate function to streamline type-checking and reuse it across both anchor and button atoms. Directly using `onClick?.(e)` is not feasible due to ESLint errors, as `onClick` is explicitly passed as a prop and requires proper type definitions. While it's possible to use an inline onClick handler like this:
```
onClick={(e: MouseEvent<HTMLButtonElement>) => {
  if (!disabled) {
    if (onClick) {
      ;(onClick as MouseEventHandler<HTMLButtonElement>)(e)
    }
  } else {
    e.preventDefault() // Explicitly prevent default if disabled
  }
}}
```
extracting the logic ensures better code reuse and consistency. Please let me know your thoughts on this approach.

Reply:
**note:** As we discussed offline, this is happening because we're not correctly narrowing all the props when we call `isButtonAsButtonElement({ href })` in our main if-statement.

We agreed this can wait until another time to improve 👍